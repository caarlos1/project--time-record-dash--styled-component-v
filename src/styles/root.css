:root {
  --color-text: #2c3e50;
  --color-background: #ffffff;
  --color-primary: #605bff;
  --section-gap: 160px;
}

*,
*::before,
*::after {
  box-sizing: border-box;
  margin: 0;
  position: relative;
  font-weight: normal;
}

html {
  font-size: 62.5%;
  line-height: 62.5%;
}

body {
  min-height: 100vh;
  max-width: 100vw;
  color: var(--color-text);
  background: var(--color-background);
  transition: color 0.5s, background-color 0.5s;
  font-size: 1.6rem;
  line-height: 1.6rem;
  font-family: "Roboto", sans-serif;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  display: flex;
  position: relative;
  padding: 0 !important;
}

a {
  text-decoration: none;
  color: var(--color-text);
}

a:hover {
  color: var(--color-primary);
}
