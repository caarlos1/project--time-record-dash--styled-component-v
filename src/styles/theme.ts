export interface ThemeType {
  colors: {
    textColor: string;
    primaryColor: string;
  };
}

export const theme: ThemeType = {
  colors: {
    textColor: "#2c3e50",
    primaryColor: "#605bff",
  },
};

export const provide = { theme };
