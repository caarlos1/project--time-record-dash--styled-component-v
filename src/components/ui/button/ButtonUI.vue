<script setup lang="ts">
import * as Styled from "./ButtonUI.styles";
import { computed } from "vue";

export type ButtonUIProps = {
  label: string;
  design?: string;
  full?: boolean;
  disabled?: boolean;
};

const props = withDefaults(defineProps<ButtonUIProps>(), {
  design: "",
  full: false,
  disabled: false,
});

const computedClass = computed(() => {
  const { disabled, full, design } = props;
  return { design, disabled, full };
});
</script>

<template>
  <Styled.Button
    v-bind="props"
    :class="{ ...computedClass, [props.design]: true }"
    data-test="button-ui"
  >
    {{ label }}
  </Styled.Button>
</template>
