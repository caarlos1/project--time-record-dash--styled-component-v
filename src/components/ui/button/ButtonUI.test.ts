import { describe, it, expect } from "vitest";
import { mount } from "@vue/test-utils";

import ButtonUI from "./ButtonUI.vue";

import { provide } from "@/styles/theme";
const mock = { label: "Little Button" };

describe("ButtonUI", () => {
  it("should render a button with correct label", () => {
    const w = mount(ButtonUI, {
      props: { label: mock.label },
      provide,
    });
    expect(w.text()).toContain(mock.label);
  });

  it("should render a button with full style", () => {
    const w = mount(ButtonUI, {
      props: { label: mock.label, full: true },
      provide,
    });
    expect(w.classes()).toContain("full");
  });

  it("should match a snapshot", () => {
    const w = mount(ButtonUI, {
      props: { label: mock.label, full: true, disabled: false },
      provide,
    });
    expect(w.element).toMatchSnapshot();
  });
});
