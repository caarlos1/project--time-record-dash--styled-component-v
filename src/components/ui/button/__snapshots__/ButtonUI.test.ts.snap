// Vitest Snapshot v1

exports[`ButtonUI > should match a snapshot 1`] = `
<button
  class="cJTTMW full full"
  data-test="button-ui"
  design=""
  disabled="false"
  full="true"
  label="Little Button"
>
  Little Button
</button>
`;
